### PR TITLE
Added KY committee scraper

### PR DESCRIPTION
**Pull request by a-cat-named-snowball**

### CODE REVIEW

File: scrapers_next/ky/committees.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        # Make the scraper more fragile by doing an extra check to
+        # make sure expected chamber and detected chamber are the same
+        if com.chamber != self.input.get("chamber"):
+            raise Exception("Unexpected chamber")
+
+        # Add members, skipping if no members
+        try:
+            members = XPath("//ul[@class='member-list']/li/a").match(self.root)
+            for member in members:
```
Comment (johnseekins) [inline scrapers_next/ky/committees.py:98]: If possible, let's keep `for` loops outside `try` blocks.

File: scrapers_next/ky/committees.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        return com
+
+
+# Senate and House committee names end with " (S)" or " (H)"
+# Other committee names have no suffix
+def strip_committee_name(committee_name):
+    is_house_committee = committee_name.endswith(" (H)")
+    is_senate_committee = committee_name.endswith(" (S)")
+    if is_house_committee or is_senate_committee:
+        return committee_name[0:-4]
```
Comment (johnseekins) [inline scrapers_next/ky/committees.py:113]: `[0:-4]` may be hard to reverse engineer in the future. A small comment on the expected format of the object you're parsing would be helpful.

File: scrapers_next/ky/committees.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        if com.chamber != self.input.get("chamber"):
+            raise Exception("Unexpected chamber")
+
+        # Add members, skipping if no members
+        try:
+            members = XPath("//ul[@class='member-list']/li/a").match(self.root)
+            for member in members:
+                add_member(com, member.text_content())
+        except Exception:
+            skip_when_no_members = False
```
[the remark this answers is not in the document]
Comment (a-cat-named-snowball) [reply scrapers_next/ky/committees.py:101]: I should have added a comment - I wasn't sure which behavior was correct. Should committees with no members be skipped like in https://github.com/openstates/openstates-scrapers/commit/cd8bb9d1fb4111e14333eee5537b07b9f332eaa6 ?

File: scrapers_next/ky/committees.py (hunk 2 above)
Comment (a-cat-named-snowball) [reply scrapers_next/ky/committees.py:115]: A comment showing the expected format has been added with https://github.com/openstates/openstates-scrapers/pull/4413/commits/009ff96e45f2c233054883a0659ccd7d7510da6a .

**Review by johnseekins — APPROVED** [review]
Good implementation. Easy to follow. Nice work.

**Review by chrisyamas — CHANGES_REQUESTED** [review]
This was a great first spatula scraper, and a very well-executed scraper in general.

Once you've consolidated the `CommitteeList` classes and created a custom exception for that case of an unexpected chamber, then I think this will be good to go.

File: scrapers_next/ky/committees.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    def process_page(self):
+        title = CSS(".committee-title > h2").match(self.root)[0].text_content()
+        parent_committee = self.input.get("parent")
+        com = get_committee_info(title, parent_committee)
+        com.add_source(self.source.url, note="committee details page")
+
+        # Make the scraper more fragile by doing an extra check to
+        # make sure expected chamber and detected chamber are the same
+        if com.chamber != self.input.get("chamber"):
+            raise Exception("Unexpected chamber")
```
[the remark this answers is not in the document]
Comment (a-cat-named-snowball) [reply scrapers_next/ky/committees.py]: Thanks, I'll rememer to use custom exceptions. Fixed by https://github.com/openstates/openstates-scrapers/pull/4413/commits/745def0014ecd60b99ac20efc5734e2e97bb3ecd

File: scrapers_next/ky/committees.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    source = "https://legislature.ky.gov/Committees/special-committee"
+    selector = CSS("[data-committeersn]")
+
+    def process_item(self, item):
+        href = item.find("a").get("href")
+        parent = get_parent(self.root, item)
+        return CommitteeDetail(
+            {"parent": parent, "chamber": "legislature"},
+            source=URL(href, timeout=30),
+        )
```
[the remark this answers is not in the document]
Comment (a-cat-named-snowball) [reply scrapers_next/ky/committees.py]: Implemented in https://github.com/openstates/openstates-scrapers/pull/4413/commits/954f105e5b79b40b3f207754dca8dca9e80c13ea

File: scrapers_next/ky/committees.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+
+# Used by CommitteeList classes to get and item's parent committee name
```
[the remark this answers is not in the document]
Comment (a-cat-named-snowball) [reply scrapers_next/ky/committees.py]: Thanks for catching that, fixed in https://github.com/openstates/openstates-scrapers/pull/4413/commits/166a54f38a82323b19031fcb4099b5e83c141736

**Review by chrisyamas — APPROVED** [review]
Thanks for making those changes, looks good!

**Pull request merged**